src/settings.py

```diff
@@ -153,6 +153,8 @@ class GlobalSettings:
     grammar_search_expected_nodes_backtracking_cost = 1e3
     grammar_search_expected_nodes_include_suspicious_score = False
     grammar_search_expected_nodes_allow_noops = True
+    # If None, defaults to CFG.max_skeletons_optimized.
+    grammar_search_max_skeletons_optimized = None
```

Comment:
instead of None should we just do `grammar_search_max_skeletons_optimized = max_skeletons_optimized`? btw our parse_args stuff might not work with `--grammar_search_max_skeletons_optimized None`, you may want to test that (update `string_to_python_object`)

Reply:
(resolved on slack: keeping it this way because otherwise, command line override of `max_skeletons_optimized` alone would not affect `grammar_search_max_skeletons_optimized`, which is confusing)